### PR TITLE
Fix misleading example of reusable workflows

### LINKED ISSUE #19904: Fix misleading example of reusable workflows
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/using-workflows/reusing-workflows

### What part(s) of the article would you like to see updated?

The [example reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows#example-reusable-workflow) in this article can lead newcomers to mix up reusable workflows with [local actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-an-action-in-the-same-repository-as-the-workflow).

A recent change from #15481 made this worse/more misleading. I propose to at least revert c9186d07c3681362b9707d6a88f8678cee8ab3b1 and maybe add a comment that the example references an action not a reusable workflow.

### Additional information

I just spent some time trying (and failing) to use a reusable workflow as a step in a workflow job because I was mislead by this example, as have [other users](https://github.com/orgs/community/discussions/27362) apparently.

### PR DESCRIPTION
**Pull request by joshjohanning**
### Why:

Closes #19904

cc @dne @lucascosti 

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Entirely removing the [local action reference](https://docs.github.com/en/actions/using-workflows/reusing-workflows#example-reusable-workflow) since with a reusable workflow, it's confusing.

The local action needs to live in the _calling_ repository, not the reusable workflow repository. 

I just used a generic/non-existent `my-org/my-action@main` action reference instead to not confuse multiple topics.

image

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

File: content/actions/using-workflows/reusing-workflows.md [hunk 1]
```diff
@@ -168,12 +168,13 @@ jobs:
     name: Pass input and secrets to my-action
     runs-on: ubuntu-latest
     steps:
-      - uses: ./.github/workflows/my-action
+      - uses: my-org/my-action@main
```
Comment (lucascosti) [inline content/actions/using-workflows/reusing-workflows.md:171]:
```suggestion
      - uses: octo-org/my-action@v1
```

File: content/actions/using-workflows/reusing-workflows.md [hunk 2]
```diff
@@ -127,7 +127,7 @@ You can define inputs and secrets, which can be passed from the caller workflow
        runs-on: ubuntu-latest
        environment: production
        steps:
-         - uses: ./.github/workflows/my-action
+         - uses: my-org/my-action@main
```
Comment (lucascosti) [inline content/actions/using-workflows/reusing-workflows.md:130]:
```suggestion
         - uses: octo-org/my-action@v1
```

**Review by lucascosti — DISMISSED** [review]
Thanks for this, @joshjohanning! I made a slight change to the action to match our usual style (e.g. using `octo-org` as an example, and using a tag as a ref). I agree that not showing a local action should make things a lot clearer. 🙇‍♂️

**Review by lucascosti — APPROVED** [review]

**Comment by joshjohanning** [comment]
@lucascosti perfect!

**Pull request merged**